solidity/contracts/libs/RateLimited.sol

```diff
@@ -88,20 +112,22 @@ contract RateLimited is OwnableUpgradeable {
 
     /**
      * Validate an amount and decreases the currentCapacity
-     * @param _newAmount The amount to consume the fill level
+     * @param _consumedAmount The amount to consume the fill level
      * @return The new filled level
      */
     function validateAndConsumeFilledLevel(
-        uint256 _newAmount
+        uint256 _consumedAmount
     ) public returns (uint256) {
         uint256 adjustedFilledLevel = calculateCurrentLevel();
-        require(_newAmount <= adjustedFilledLevel, "RateLimitExceeded");
+        require(_consumedAmount <= adjustedFilledLevel, "RateLimitExceeded");
 
         // Reduce the filledLevel and update lastUpdated
-        uint256 _filledLevel = adjustedFilledLevel - _newAmount;
+        uint256 _filledLevel = adjustedFilledLevel - _consumedAmount;
         filledLevel = _filledLevel;
         lastUpdated = block.timestamp;
 
+        emit ConsumedFilledLevel(filledLevel, lastUpdated);
```

Comment:
whats the intended use for this?

Reply:
Part of the audit, to have events for state changes.

Reply:
personally dont think this applies pervasively but okay